### PR TITLE
Add empty endpoints for the Alert resource

### PR DESCRIPTION
**Pull request by dluc**
This change is [Reviewable](https://reviewable.io/reviews/azure/device-telemetry-java/23)

### CODE REVIEW

**Comment by dluc** [comment]
Ready for CR

---

Review status: 0 of 15 files reviewed at latest revision, 3 unresolved discussions.

---

*[DEVELOPMENT.md, line 27 at r1](https://reviewable.io:443/reviews/azure/device-telemetry-java/23#-KnqXDVIA-sSEE4ujnUC:-KnuVv64SXIZt_bjv5od:b-7ql1td) ([raw file](https://github.com/azure/device-telemetry-java/blob/46d1f3fbdccec7c7182b60d855fc46937cf29f34/DEVELOPMENT.md#L27)):*
<details><summary><i>Previously, jillcary (Jill Bender) wrote…</i></summary><blockquote>

nit: "you"
</blockquote></details>

fixed

---

*[DEVELOPMENT.md, line 30 at r1](https://reviewable.io:443/reviews/azure/device-telemetry-java/23#-KntwomRoYQOJPXkAL6-:-KnuVyZvR8aBY_oWAXl-:b1kxy83) ([raw file](https://github.com/azure/device-telemetry-java/blob/46d1f3fbdccec7c7182b60d855fc46937cf29f34/DEVELOPMENT.md#L30)):*
<details><summary><i>Previously, jillcary (Jill Bender) wrote…</i></summary><blockquote>

> `compile`: compile the project.

It doesn't appear that we have a compile script at the moment in the repo. 
</blockquote></details>

good catch, fixed

---

*[app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java, line 30 at r1](https://reviewable.io:443/reviews/azure/device-telemetry-java/23#-KnqWfqKaSNGqXegciMo:-KnqZGjaj-FpopsntQrD:bsfgth5) ([raw file](https://github.com/azure/device-telemetry-java/blob/46d1f3fbdccec7c7182b60d855fc46937cf29f34/app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java#L30)):*
<details><summary><i>Previously, jillcary (Jill Bender) wrote…</i></summary><blockquote>

Why do we need to get a single alert for a rule? Is this covered by the alerts controller? 

Or does this return all of the alerts for a specific rule? (comment is confusing if that's the case)
</blockquote></details>

forgot to update the comment after creating the method

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/device-telemetry-java/23#-:-KnuW4nS788Jy_5XtpC5:b-yuye54)*

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -3,25 +3,106 @@
...
+* "Open" the project with IntelliJ, the IDE should automatically recognize the
+  SBT structure. Wait for the IDE to download some dependencies (see IntelliJ
+  status bar)
+* Create a new Run Configuration, of type "SBT Task" and enter "run 9004"
+  (including the double quotes). This ensures that the service to start using
+  the TCP port 9004
+* Either from the toolbar or the Run menu, execute the configuration just
+  created, using the Debug command/button
+* Test that the service is up and running pointing your browser to
+  http://127.0.0.1:9004/v1/status
```
Comment (timlaverty) [inline DEVELOPMENT.md]: service name?

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers;
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java]: The project name focuses on reading Telemetry but the service will support CRUD for Rules & Alerts definitions and Read/Update for the Alert Status table.  Should the project/service be updated so it's more clear what it offers?

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+public class AlertsController {
+    /**
+     * Return a list of alerts. The list of alerts can be paginated, and
+     * filtered by device, period of time, status. The list is sorted
+     * chronologically, by default starting from the oldest alert, and
+     * optionally from the most recent.
+     *
+     * @return List of alerts.
+     */
+    public Result list() {
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsController.java]: Assuming you'll want paging and total count as the # of alerts could get big.

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import play.mvc.Result;
+
+import static play.libs.Json.toJson;
+import static play.mvc.Results.ok;
+
+// TODO: Review and see if we can either extend the Alert API or the Rules API.
+public class AlertsByRuleController {
```
Comment (timlaverty) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java]: maybe need paging?  Can we estimate the # of alerts per rule?  The parameters are operator defined so could be very chatty.

File: DEVELOPMENT.md (hunk 1 above)
Comment (dluc) [inline DEVELOPMENT.md]: I'm not sure I understand the comment, is the service name missing or...?

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java (hunk 2 above)
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java]: I agree, we need to rename the project at some point

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java (hunk 4 above)
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsByRuleController.java]: yes there are going to be query parameters to optimize the payload size. We have some [tasks](https://github.com/Azure/device-telemetry-java/issues) to work on that

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsController.java (hunk 3 above)
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/AlertsController.java]: yes, see comment above

**Pull request merged**